File: 0073-set-matrix-zeroes/solution.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        const int m = matrix[0].size();
        const int n = matrix.size();
        
        vector<pair<int,int>> vp ;
        
        for(int i = 0 ; i < n ; i++){
            for(int j = 0 ; j < m ; j++){
                if(!matrix[i][j])
                    vp.push_back({i,j});
            }
        }
        
        for(auto &k : vp){
            //cout << k.first << " " << k.second ;
            for(int i = 0 ; i < m ; i++ )
                matrix[k.first][i] = 0 ;
            
            for(int i = 0 ; i < n ; i++ )
                matrix[i][k.second] = 0 ;
        }
            
    }
};

File: 0073-set-matrix-zeroes/solution.java
class Solution {
    class Pair{
        int x ;
        int y ;
            Pair(int x , int y){
                this.x = x;
                this.y = y ;
            }
    }

    public void setZeroes(int[][] mat) {
    List<Pair> points = new ArrayList<>();
        int m = mat.length;
        int n = mat[0].length;
        for(int i = 0; i < m ; i++){
            for(int j = 0 ; j < n ; j++){
                if(mat[i][j] == 0){
                    Pair p = new Pair(i,j);
                    points.add(p);
                }
            }
        }

        for(int i = 0 ; i < points.size() ; i++){
            int x = points.get(i).x;
            for(int j = 0 ; j < n ; j++)
                mat[x][j] = 0 ;
        }

        for(int i = 0 ; i < points.size() ; i++){
            int y = points.get(i).y;
            for(int j = 0 ; j < m ; j++)
                mat[j][y] = 0 ;
        }
    }
}
